src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +72,6 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public abstract boolean isMutating ();
 }
```

Comment:
Try to give abstract methods a default behaviour and not just to define the method. For instance, returning `false`, the override the method in child classes wherever appropriate.